### PR TITLE
fix: Reset the stepper Jacobian in the KF smoothing

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
This PR solves the low Pt issue that was occurring when using the KF and the smoothing. The Jacobian of the stepper is now reset before performing the last propagation from the last state to the perigee after the smoothing. 

This is the equivalent of replacing the `stepper.update` by a `stepper.resetState` but this implementation allows us to avoid computing the free parameters twice.

### CODE REVIEW

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 1]
```diff
@@ -1111,12 +1111,20 @@ class CombinatorialKalmanFilter {
                        firstCreatedState.smoothedCovariance(),
                        firstCreatedState.referenceSurface());
         reverseDirection = (firstIntersection.intersection.pathLength < 0);
+        // Reinitialize the stepping jacobian
+        state.stepping.jacToGlobal =
```
Comment (XiaocongAi) [inline Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:1115]: After a second thought, I'm not sure if we really need to reset the `jabToGlobal` after calling the stepper `update` as the `jacToGlobal` is already reset there.

File: Core/include/Acts/TrackFitting/KalmanFitter.hpp [hunk 2]
```diff
@@ -965,12 +965,20 @@ class KalmanFitter {
                        firstCreatedState.smoothedCovariance(),
                        firstCreatedState.referenceSurface());
         reverseDirection = (firstIntersection.intersection.pathLength < 0);
+        // Reinitialize the stepping jacobian
+        state.stepping.jacToGlobal =
```
Comment (XiaocongAi) [inline Core/include/Acts/TrackFitting/KalmanFitter.hpp:969]: Same here. Do we need to reset the `jacToGlobal` here?

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp (hunk 1 above)
Comment (Corentin-Allaire) [reply Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:1115]: Your right, it is already reset in the `update` I will remove it

**Review by XiaocongAi — APPROVED** [review]

**Comment by paulgessinger** [comment]
Root output hashes are still changed. Can you look at that @Corentin-Allaire?

**Comment by Corentin-Allaire** [comment]
@paulgessinger I have updated the hashes.

**Comment by Corentin-Allaire** [comment]
@XiaocongAi could you reapprove ? I had to update the hashes of the python test

**Pull request merged**